app/routes/createObservation.tsx

```diff
@@ -0,0 +1,60 @@
+import { json } from "@remix-run/node";
+import { createObservation } from "~/utils/db.server";
+import { createObservation as uploadToS3, genKey } from "~/utils/s3.server";
+
+type Location = { x: number; y: number };
+
+export const action = async ({ request }: { request: Request }) => {
+  if (request.method === "POST") {
+    const formData = await request.json();
+
+    const {
+      title,
+      description,
+      location,
+      picture,
+    }: {
+      title: string;
+      description: string;
+      location: Location;
+      picture: { data: string; ext: string } | null;
```

Comment:
The data won't be a string, right? It's going to be an image so some form of binary data. REST APIs aren't all that great for transferring binary data so you should change that. It's better to use something like "multipart/form-data" here instead of JSON.

Reply:
Should be correct file validation now.